### PR TITLE
Fix experiment run service as per new audit log design

### PR DESCRIPTION
**Pull request by anandjakhaniya**

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunServiceImpl.java [hunk 1]
```diff
@@ -69,20 +68,28 @@ public ExperimentRunServiceImpl(ServiceSet serviceSet, DAOSet daoSet) {
   }
 
   private void saveAuditLogs(
-      UserInfo userInfo, String action, List<String> resourceIds, String metadataBlob) {
+      Optional<UserInfo> userInfo,
+      ModelDBServiceActions action,
+      List<String> resourceIds,
+      String request,
+      String response,
+      Long workspaceId) {
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunServiceImpl.java:77]: Same comment from dataset version.

**Review by conradoverta — APPROVED** [review]

**Review by conradoverta — CHANGES_REQUESTED** [review]
Pressed the wrong button

File: backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunServiceImpl.java (hunk 1 above)
Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunServiceImpl.java:77]: https://github.com/VertaAI/modeldb/pull/1979

**Review by conradoverta — APPROVED** [review]

File: backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunServiceImpl.java [hunk 2]
```diff
@@ -69,20 +69,28 @@ public ExperimentRunServiceImpl(ServiceSet serviceSet, DAOSet daoSet) {
...
                         resourceId,
-                        ModelDBConstants.EXPERIMENT_RUN,
-                        Service.MODELDB_SERVICE.name(),
-                        metadataBlob))
+                        ModelDBServiceResourceTypes.EXPERIMENT_RUN,
+                        Service.MODELDB_SERVICE,
+                        MonitoringInterceptor.METHOD_NAME.get(),
+                        request,
+                        response,
+                        workspaceId))
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunServiceImpl.java]: Anti-pattern?

File: backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunServiceImpl.java [hunk 3]
```diff
@@ -1988,13 +2063,35 @@ public void deleteExperimentRuns(
...
+
+      UserInfo currentLoginUserInfo = authService.getCurrentLoginUserInfo();
+      for (Entry<String, Long> entry : workspaceIdByExperimentRunId.entrySet()) {
+        saveAuditLogs(
+            Optional.of(currentLoginUserInfo),
+            ModelDBServiceActions.DELETE,
+            Collections.singletonList(entry.getKey()),
+            ModelDBUtils.getStringFromProtoObject(request),
+            ModelDBUtils.getStringFromProtoObject(response),
+            entry.getValue());
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunServiceImpl.java]: Multiple log entries. It has to be one per request.

**Review by conradoverta — CHANGES_REQUESTED** [review]

**Review by conradoverta — APPROVED** [review]

**Pull request merged**